### ISSUE TITLE
Can't infer correct NuxtApp property type after registering @nuxtjs/color-mode

### ISSUE THREAD

**Issue by danielroe** [nuxt-modules/color-mode#138]
Originally posted in https://github.com/nuxt/nuxt.js/issues/13787

---

### Environment

- Operating System: `Windows_NT`
- Node Version:     `v16.14.2`
- Nuxt Version:     `3.0.0-rc.1`
- Package Manager:  `npm@8.7.0`
- Builder:          `vite`
- User Config:      `modules`
- Runtime Modules:  `@nuxtjs/color-mode@3.0.2`
- Build Modules:    `-`

### Reproduction

https://stackblitz.com/edit/github-yq4qz8

### Describe the bug

When using a property provided by a custom plugin, it can't infer correct NuxtApp property type that always shows `any` after registering `@nuxtjs/color-mode`. 

But it works when the property used in template alone.

Please try the demo in VS Code.

**Comment by danielroe** [comment]
https://github.com/P4sca1/nuxt-plugin-provide-types

**Comment by P4sca1** [comment]
@danielroe I found out that removing the following lines resolves the issue.
https://github.com/nuxt-modules/color-mode/blob/f5253feb5a547a61ee38edca243947cf1b20c58c/src/runtime/types.ts#L28-L30

Changing them to the following also works (and might be needed for Bridge typings?):
```ts
declare module '#app' {
    interface NuxtApp extends PluginInjection {
        $colorMode: ColorModeInstance;
    }
}
```
Maybe adding `$colorMode` to the `PluginInjection` interface was just forgotten?
Could you elaborate which interfaces are used by Nuxt Bridge / Nuxt 3 for TypeScript typings?

According to [the docs](), the following declaration would be sufficient for Nuxt 3:
```ts
export interface ColorModeInstance {
    preference: string;
    value: string;
    unknown: boolean;
    forced: boolean;
}

declare module '#app' {
  interface NuxtApp {
    $colorMode: ColorModeInstance
  }
}

declare module '@vue/runtime-core' {
  interface ComponentCustomProperties {
    $colorMode: ColorModeInstance
  }
}

export { }
```

I don't really know what all the other definitions are for in the `types.d.ts` file.

**Comment by P4sca1** [comment]
@danielroe @pi0 Could you have a look at this? This issue really kills the DX in Nuxt 3.

**Comment by danielroe** [comment]
This seems to be working fine for me using your reproduction + latest color mode + rc113.

<img width="482" alt="CleanShot 2022-11-11 at 18 45 13@2x" src="https://user-images.githubusercontent.com/28706372/201409769-8a0439ad-4651-40d6-8bfc-2b7b1caaeedb.png">

```json
  "devDependencies": {
    "@nuxtjs/color-mode": "latest",
    "nuxt": "3.0.0-rc.13"
  }
```

Are you still experiencing this?

**Comment by P4sca1** [comment]
@danielroe I upgraded the dependencies in https://github.com/P4sca1/nuxt-plugin-provide-types and still experience the issue.

However I noticed the following: When you just loaded VSCode, or restarted the Vue server (Volar: Restart Vue Server), the correct typing for `$test` will be there for a few seconds. After a few seconds, when you reopen the file `app.vue`, the typings will be gone and `$test` will be typed `any`.
